core/src/main/java/org/apache/iceberg/SchemaUpdate.java

```diff
@@ -438,13 +438,18 @@ private static Schema applyChanges(Schema schema, List<Integer> deletes,
                                      Multimap<Integer, Types.NestedField> adds,
                                      Multimap<Integer, Move> moves,
                                      Set<String> identifierFieldNames) {
-    // validate existing identifier fields are not deleted
+    Map<Integer, Integer> idToParent = TypeUtil.indexParents(schema.asStruct());
+
     for (String name : identifierFieldNames) {
       Types.NestedField field = schema.findField(name);
       if (field != null) {
-        Preconditions.checkArgument(!deletes.contains(field.fieldId()),
-            "Cannot delete identifier field %s. To force deletion, " +
-                "also call setIdentifierFields to update identifier fields.", field);
+        Integer idNotToDelete = field.fieldId();
```

Comment:
Can we keep the original Precondition check as is, and then have code after to check parents.  

The message can have more information in the new case, and the code is clearer
```
 Preconditions.checkArgument(!deletes.contains(field.fieldId()),
            "Cannot delete identifier field %s. To force deletion, " +
                "also call setIdentifierFields to update identifier fields.", field);
Integer parentId = fieldId;
while (parentId != null) {
   Preconditions.checkArgument(... "Cannot delete field %s as it will delete nested identifier field %s",...)
   parentId = idToParent.get(parentId);
}
```

Reply:
Will update.